src/shims/backtrace.rs

```diff
@@ -49,31 +69,87 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             })
             .collect();
 
-        let len = ptrs.len();
+        let len = ptrs.len().try_into().unwrap();
```

Comment:
This has some very non-local type inference, would be good to state the desired type of `len` on this line.